### ISSUE TITLE
Make `RuleParser` a Builder

### ISSUE THREAD

**Issue by mmarx** [knowsys/nemo#60]
`RuleParser` should have a `finalize` method that moves out the parsed Data, Program, and Dictionary.

Depends on #58 being merged.

**Comment by mmarx** [comment]
Reevaluate if this is still necessary after removing the dictionary from the logical layer.

**Comment by monsterkrampe** [comment]
Is this still an issue?
If so, should we schedule it for an upcoming release?

**Comment by mmarx** [comment]
Closing in favour of #59.
